### PR TITLE
configure model from dataset

### PR DESCRIPTION
**Pull request by blanchco**
# Description

* create model configurations from dataset with gollm

https://github.com/DARPA-ASKEM/terarium/assets/33158416/e5778724-71fa-4f74-b48f-c74377d50a7e

Resolves #(issue)

### CODE REVIEW

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/tasks/ConfigureModelResponseHandler.java [hunk 1]
```diff
@@ -69,20 +70,24 @@ public void onSuccess(TaskResponse resp) {
...
-							parameter.setValue(conditionParameter.get("value").doubleValue());
-						}
-					});
-				});
+            // For each configuration, create a new model configuration with parameters set
+            for (JsonNode condition : configurations.response.get("conditions")) {
+                // Map the parameters values to the model
+                final Model modelCopy = new Model(model);
+                 List<ModelParameter> modelParameters;
+					if(modelCopy.getHeader().getSchemaName().toLowerCase().equals("regnet")) {
```
Comment (dvince2) [inline packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/tasks/ConfigureModelResponseHandler.java:78]: Is this a weird github thing or is there wonky indenting here?

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/gollm/GoLLMController.java [hunk 2]
```diff
@@ -234,6 +234,7 @@ public ResponseEntity<TaskResponse> createConfigFromDatasetTask(
 				}
 			}
 
+			log.info("Datasets:" + datasets);
```
Comment (dvince2) [inline packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/gollm/GoLLMController.java:237]: logging

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/ModelController.java [hunk 3]
```diff
@@ -295,9 +299,37 @@ ResponseEntity<List<ModelConfiguration>> getModelConfigurationsForModelId(
 						log.error("Unable to get the document " + documentId, e);
 					}
 				});
+
+				// Find the Dataset Assets linked via provenance to the model configuration
+				final ProvenanceQueryParam datasetSearchBody = new ProvenanceQueryParam();
+				datasetSearchBody.setRootId(config.getId());
+				datasetSearchBody.setRootType(ProvenanceType.MODEL_CONFIGURATION);
+				datasetSearchBody.setTypes(List.of(ProvenanceType.DATASET));
+				final Set<String> datasetIds = provenanceSearchService.modelConfigFromDataset(datasetSearchBody);
```
Comment (dvince2) [inline packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/ModelController.java:308]: This method was written to throw a `IllegalArgumentException` in some cases - if thats going to be an expected path then we should throw some/all of this in a try/catch to handle that failure case... but if its a super rarety and thats just a bit of safety code its probably fine!

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/tasks/ConfigureModelResponseHandler.java (hunk 1 above)
Comment (blanchco) [reply packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/tasks/ConfigureModelResponseHandler.java:78]: it was a little off, but githubs making it look a lot worse 😆 
Screenshot 2024-03-01 at 9 43 00 AM

Just fixed that up

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/tasks/TaskService.java [hunk 4]
```diff
@@ -214,6 +214,8 @@ void onTaskResponseOneInstanceReceives(final Message message) {
 			}
 
 			try {
+				log.info("Response handlers:" + responseHandlers.toString());
```
[the remark this answers is not in the document]
Comment (blanchco) [reply packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/tasks/TaskService.java:217]: those pesky debug logs 🐛 , removed

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/ModelController.java (hunk 3 above)
Comment (blanchco) [reply packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/ModelController.java:308]: Good idea, I've make the catch for all exceptions now!

File: packages/client/hmi-client/src/workflow/ops/model-config/tera-model-config.vue [hunk 5]
```diff
@@ -691,9 +732,12 @@ const onSelection = (id: string) => {
 
 const fetchConfigurations = async (modelId: string) => {
 	if (modelId) {
-		loadingConfigs.value = true;
-		suggestedConfirgurationContext.value.tableData = await getModelConfigurations(modelId);
-		loadingConfigs.value = false;
+		// FIXME: since configurations are made on the backend on the fly, we need to wait for the db to update before fetching, here's an artificaial delay
```
Comment (dvince2) [inline packages/client/hmi-client/src/workflow/ops/model-config/tera-model-config.vue:735]: Off the top of my head I don't have a _better_ solution that isn't adding a bunch of work - polling, mq, etc - but this is something that should for sure at minimum have a high priority issue opened up today to address as soon as the hackathon is over! At best the user is going to have to wait an extra ~second here, but at worse there isn't a guarantee that 800ms would be long enough then we're back to where we were before.

**Review by dvince2 — COMMENTED** [review]
I'm going to defer to @YohannParis on the front end sleep call here - back end looks fine!

File: packages/client/hmi-client/src/services/goLLM.ts [hunk 6]
```diff
@@ -54,7 +54,33 @@ export async function configureModel(documentId: string, modelId: string): Promi
...
+	}
+}
+
+export async function configureModelFromDatasets(modelId: string, datasetIds: string[]) {
+	try {
+		// FIXME: Using first dataset for now...
+		const response = await API.post<TaskResponse>('/gollm/configure-from-dataset', null, {
+			params: {
+				'model-id': modelId,
+				'dataset-ids': datasetIds[0]
```
Comment (YohannParis) [inline packages/client/hmi-client/src/services/goLLM.ts:67]: Try the following before changing from params to body
```suggestion
				'dataset-ids': datasetIds.join()
```

File: packages/client/hmi-client/src/workflow/ops/model-config/tera-model-config.vue [hunk 7]
```diff
@@ -19,14 +19,14 @@
 								>({{ suggestedConfirgurationContext.tableData.length }})</span
 							>
 							<Button
-								outlined
-								label="Extract configurations from a document"
-								size="small"
 								icon="pi pi-cog"
-								@click.stop="extractConfigurations"
-								:disabled="loadingConfigs || !documentId || !model.id"
+								label="Extract Configurations"
```
Comment (YohannParis) [inline packages/client/hmi-client/src/workflow/ops/model-config/tera-model-config.vue:23]:
```suggestion
								label="Extract Configurations from inputs"
```

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/ModelController.java [hunk 8]
```diff
@@ -275,11 +279,11 @@ ResponseEntity<List<ModelConfiguration>> getModelConfigurationsForModelId(
 
 
 				// Find the Document Assets linked via provenance to the model configuration
-				final ProvenanceQueryParam body = new ProvenanceQueryParam();
-				body.setRootId(config.getId());
-				body.setRootType(ProvenanceType.MODEL_CONFIGURATION);
-				body.setTypes(List.of(ProvenanceType.DOCUMENT));
-				final Set<String> documentIds = provenanceSearchService.modelConfigFromDocument(body);
+				final ProvenanceQueryParam documentSearchBody = new ProvenanceQueryParam();
```
Comment (YohannParis) [inline packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/ModelController.java:282]:
```suggestion
				final ProvenanceQueryParam queryParam = new ProvenanceQueryParam();
```

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/tasks/ConfigureFromDatasetResponseHandler.java [hunk 9]
```diff
@@ -70,26 +71,28 @@ public void onSuccess(TaskResponse resp) {
...
-					conditionParameters.forEach((conditionParameter) -> {
-						if (parameter.getId().equals(conditionParameter.get("id").asText())) {
-							parameter.setValue(conditionParameter.get("value").doubleValue());
-						}
-					});
-				});
+                // Map the parameters values to the model
+                final Model modelCopy = new Model(model);
+                List<ModelParameter> modelParameters;
+					if(modelCopy.getHeader().getSchemaName().toLowerCase().equals("regnet")) {
```
Comment (YohannParis) [inline packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/tasks/ConfigureFromDatasetResponseHandler.java:77]: Could you make a method `isRegNet` in the Model.model? With @dvince2

**Review by YohannParis — APPROVED** [review]

File: packages/client/hmi-client/src/workflow/ops/model-config/tera-model-config.vue (hunk 5 above)
Comment (blanchco) [reply packages/client/hmi-client/src/workflow/ops/model-config/tera-model-config.vue:741]: @kbirk is looking into adding a way to get a finished message after BE is done processing

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/ModelController.java (hunk 8 above)
Comment (blanchco) [reply packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/ModelController.java:282]: changed to `documentQueryParams` and `datasetQueryParams`

File: packages/client/hmi-client/src/services/goLLM.ts (hunk 6 above)
Comment (blanchco) [reply packages/client/hmi-client/src/services/goLLM.ts:67]: this worked 👍

**Pull request merged**